CMakeLists.txt

```diff
@@ -222,19 +219,16 @@ if(HIP_PLATFORM STREQUAL "hcc")
       target_link_libraries(hip_hcc PRIVATE hc_am)
       target_link_libraries(hip_hcc_static PRIVATE hc_am)
     endif()
-    add_library(hip_device STATIC ${SOURCE_FILES_DEVICE})
 
     string(REPLACE " " ";" HCC_CXX_FLAGS_LIST ${HCC_CXX_FLAGS})
-    foreach(TARGET hip_hcc hip_hcc_static hip_device)
+    foreach(TARGET hip_hcc hip_hcc_static)
         target_include_directories(${TARGET} SYSTEM INTERFACE $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include>;${HSA_PATH}/include)
     endforeach()
     add_library(host INTERFACE)
-    target_link_libraries(host INTERFACE hip_hcc)
-    add_library(device INTERFACE)
     if(HIP_COMPILER STREQUAL "hcc")
-        target_link_libraries(device INTERFACE host hip_device hcc::hccrt hcc::hc_am)
-    elseif(HIP_COMPILER STREQUAL "clang")
-        target_link_libraries(device INTERFACE host hip_device)
+        target_link_libraries(host INTERFACE hip_hcc hcc::hccrt hcc::hc_am)
```

Comment:
`hcc::hccrt` and `hcc::hc_am` require hcc compiler, but `hip::host` should work when compiling with gcc as well.